sky/utils/controller_utils.py

```diff
@@ -187,79 +187,48 @@ def from_type(cls, controller_type: str) -> Optional['Controllers']:
-            (clouds.Lambda, clouds.SCP, clouds.Fluidstack, clouds.Paperspace)):
-            # no need to install any cloud dependencies for lambda, scp,
-            # fluidstack and paperspace
-            continue
-        if isinstance(cloud, clouds.AWS):
-            step_prefix = prefix_str.replace('<step>',
-                                             str(len(setup_clouds) + 1))
-            commands.append(f'echo -en "\\r{step_prefix}AWS{empty_str}" && ' +
-                            aws_dependencies_installation)
-            setup_clouds.append(str(cloud))
-        elif isinstance(cloud, clouds.Azure):
-            step_prefix = prefix_str.replace('<step>',
-                                             str(len(setup_clouds) + 1))
-            commands.append(
-                f'echo -en "\\r{step_prefix}Azure{empty_str}" && '
-                'pip list | grep azure-cli > /dev/null 2>&1 || '
-                'pip install "azure-cli>=2.31.0" azure-core '
-                '"azure-identity>=1.13.0" azure-mgmt-network > /dev/null 2>&1')
-            # Have to separate this installation of az blob storage from above
-            # because this is newly-introduced and not part of azure-cli. We
-            # need a separate installed check for this.
-            commands.append(
-                'pip list | grep azure-storage-blob > /dev/null 2>&1 || '
-                'pip install azure-storage-blob msgraph-sdk > /dev/null 2>&1')
-            setup_clouds.append(str(cloud))
-        elif isinstance(cloud, clouds.GCP):
-            step_prefix = prefix_str.replace('<step>',
-                                             str(len(setup_clouds) + 1))
+        cloud_python_dependencies: List[str] = dependencies.extras_require[
+            cloud.canonical_name()]
+
+        if isinstance(cloud, clouds.Azure):
+            # azure-cli cannot be normally installed by uv.
+            # See comments in sky/skylet/constants.py.
+            cloud_python_dependencies.remove(dependencies.AZURE_CLI)
+
+            step_prefix = prefix_str.replace('<step>', str(len(commands) + 1))
             commands.append(
-                f'echo -en "\\r{step_prefix}GCP{empty_str}" && '
-                'pip list | grep google-api-python-client > /dev/null 2>&1 || '
-                'pip install "google-api-python-client>=2.69.0" '
+                f'echo -en "\\r{step_prefix}azure-cli{empty_str}" &&'
+                f'uv pip install --prerelease=allow "{dependencies.AZURE_CLI}" '
                 '> /dev/null 2>&1')
-            # Have to separate the installation of google-cloud-storage from
-            # above because for a VM launched on GCP, the VM may have
-            # google-api-python-client installed alone.
-            commands.append(
-                'pip list | grep google-cloud-storage > /dev/null 2>&1 || '
-                'pip install google-cloud-storage > /dev/null 2>&1')
-            commands.append(f'{gcp.GOOGLE_SDK_INSTALLATION_COMMAND}')
```

Comment:
Do we still have the google cloud SDK installed in the new setting? Could we test this PR with clean env and new controller?

Reply:
Ah thanks, good catch.

Reply:
Confirmed GCP broken with this change. Fixed in the latest version.